### PR TITLE
release: add CRDNotFound condition

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/5278.

- Add CRDNotFound condition.
- Rename SecretNotExist to SecretNotFound.
- Use waitError instead of stillExistsError.

### CODE REVIEW

File: pkg/release/spec.go [hunk 1]
```diff
@@ -12,13 +12,16 @@ type ConditionSet interface {
...
 	// Kubernetes API described by the given label selector.
 	PodExists(ctx context.Context, namespace, labelSelector string) ConditionFunc
 	// SecretExists returns a function waiting for the Secret to appear in the
 	// Kubernetes API described by the given name.
 	SecretExists(ctx context.Context, namespace, name string) ConditionFunc
-	// SecretNotExist returns a function waiting for the Secret to
+	// SecretNotFound returns a function waiting for the Secret to
 	// disappear in the Kubernetes API described by the given name.
-	SecretNotExist(ctx context.Context, namespace, name string) ConditionFunc
+	SecretNotFound(ctx context.Context, namespace, name string) ConditionFunc
```
Comment (kopiczko) [inline pkg/release/spec.go]: This is only used in aws-operator. I'll update that right away.

**Review by xh3b4sd — APPROVED** [review]
Fine with that when the build succeeds.

**Comment by kopiczko** [comment]
Reason for failing CI:

```
Removing intermediate container 37529aa96f7e
Step 7/18 : RUN apk -Uuv add --update --no-cache       bash=4.4.19-r1       build-base=0.5-r1       git       jq=1.6_rc1-r1       less=530-r0       libffi-dev=3.2.1-r4       openssh-client=7.7_p1-r3       openssl       sudo=1.8.23-r2       iptables=1.6.2-r0
 ---> Running in 2b00bfdeec40
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  openssh-client-7.7_p1-r4:
    breaks: world[openssh-client=7.7_p1-r3]
The command '/bin/sh -c apk -Uuv add --update --no-cache       bash=4.4.19-r1       build-base=0.5-r1       git       jq=1.6_rc1-r1       less=530-r0       libffi-dev=3.2.1-r4       openssh-client=7.7_p1-r3       openssl       sudo=1.8.23-r2       iptables=1.6.2-r0' returned a non-zero code: 1
```

**Comment by kopiczko** [comment]
#194 should fix the build.

**Comment by kopiczko** [comment]
Green. Going.

**Pull request merged**